.github/workflows/code_size_comment.yml

```diff
@@ -0,0 +1,55 @@
+name: Code size comment
+
+on:
+  workflow_run:
+    workflows: [Check code size]
+    types: [completed]
+
+jobs:
+  comment:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: 'Download artifact'
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const fs = require('fs');
+
+            const allArtifacts = await github.rest.actions.listWorkflowRunArtifacts({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               run_id: context.payload.workflow_run.id,
+            });
+
+            const matchArtifact = allArtifacts.data.artifacts.filter((artifact) => {
+              return artifact.name == "code-size-report"
+            })[0];
+
+            const download = await github.rest.actions.downloadArtifact({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               artifact_id: matchArtifact.id,
+               archive_format: 'zip',
+            });
+
+            fs.writeFileSync(`${process.env.GITHUB_WORKSPACE}/code-size-report.zip`, Buffer.from(download.data));
+      - name: 'Unzip artifact'
+        run: unzip code-size-report.zip
+      - name: Post comment to pull request
+        uses: actions/github-script@v6
+        with:
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          script: |
+            const fs = require('fs');
+
+            github.rest.issues.createComment({
```

Comment:
It looks like this creates a comment even if there is no code size change.  IMO it would be better to only comment if the code size changes (ie increases or decreases).

Reply:
Can it be done in the other workflow, and only upload the artifact if the diff is non-zero?  Then this workflow just doesn't do anything if it can't find the artifact.

Reply:
> Then this workflow just doesn't do anything if it can't find the artifact.

This part is already implemented to handle push vs. pull request.

> Can it be done in the other workflow, and only upload the artifact if the diff is non-zero? 

I'm sure it can be done, but I've already spend too much time on this.